client/blocks/inline-help/popover.jsx

```diff
@@ -168,18 +199,51 @@ class InlineHelpPopover extends Component {
 }
 
 function mapStateToProps( state ) {
+	const siteId = getSelectedSiteId( state );
+	const currentRoute = getCurrentRoute( state );
+	const classicRoute = currentRoute.replace( '/gutenberg/', '' );
+
 	return {
 		searchQuery: getSearchQuery( state ),
 		selectedSite: getHelpSelectedSite( state ),
 		selectedResult: getInlineHelpCurrentlySelectedResult( state ),
+		selectedEditor: getSelectedEditor( state, siteId ),
+		classicUrl: `/${ classicRoute }`,
+		siteId,
 	};
 }
 
-export default connect(
-	mapStateToProps,
-	{
+function mapDispatchToProps( dispatch ) {
+	return {
+		optOut: siteId => {
+			dispatch(
+				withAnalytics(
+					composeAnalytics(
+						recordGoogleEvent(
+							'Gutenberg Opt-Out',
+							'Clicked "Switch to the classic editor" in the help popover.',
+							'Opt-In',
+							false
+						),
+						recordTracksEvent( 'calypso_gutenberg_opt_in', {
+							opt_in: false,
+						} ),
+						bumpStat( 'gutenberg-opt-in', 'Calypso Help Opt Out' )
+					),
+					setSelectedEditor( siteId, 'classic' )
+				)
+			);
+		},
 		recordTracksEvent,
 		selectResult,
 		resetContactForm: resetInlineHelpContactForm,
-	}
+		redirect: classicUrl => {
+			dispatch( navigate( classicUrl ) );
```

Comment:
We might also want to consider replacing history here, since clicking back will try to redirect to the appropriate editor
```
/** * Replaces the current url and modifies the browser history entry. Equivalent to window.replaceHistory
 * @param  {String} path Navigation path
 * @return {Object}      Action object
 */
export const replaceHistory = path => ( { type: HISTORY_REPLACE, path } );
```